plugin/pkg/scheduler/scheduler.go

```diff
@@ -183,6 +197,42 @@ func (sched *Scheduler) schedule(pod *v1.Pod) (string, error) {
 	return host, err
 }
 
+func (sched *Scheduler) preempt(preemptor *v1.Pod, scheduleErr error) (string, error) {
+	if !utilfeature.DefaultFeatureGate.Enabled(features.PodPriority) {
+		glog.V(3).Infof("Pod priority feature is not enabled. No preemption is performed.")
+		return "", nil
+	}
+	preemptor, err := sched.config.PodPreemptor.GetUpdatedPod(preemptor)
+	if err != nil {
+		glog.Errorf("Error getting the updated preemptor pod object: %v", err)
+		return "", err
+	}
+	nodeName, victims, err := sched.config.Algorithm.Preempt(preemptor, sched.config.NodeLister, scheduleErr)
```

Comment:
We shouldn't use nodeName here - we should simply return *Node object. Then we can compare with nil which is always cleaner then using len() == 0.

Reply:
I agree. I wanted it to be consistent with `Schedule`. Anyway, I changed it to return *Node.